File: src/main/java/me/sargunvohra/leveluphp/Resources.java
package me.sargunvohra.leveluphp;

import lombok.experimental.UtilityClass;
import net.minecraft.util.ResourceLocation;

/** Utility class to generate ResourceLocations for this mod. */
@UtilityClass
public class Resources {

  public final ResourceLocation texIcons = get("textures/gui/icons.png");

  public ResourceLocation get(String name) {
    return new ResourceLocation(LevelUpHp.MOD_ID, name);
  }
}

File: src/main/java/me/sargunvohra/leveluphp/Resources.kt
package me.sargunvohra.leveluphp

import net.minecraft.util.ResourceLocation

object Resources {
    val texIcons = get("textures/gui/icons.png")
    val dataGeneral = get("${LevelUpHp.MOD_ID}/general.json")
    val soundLevelUp = get("levelup")

    fun get(name: String) = ResourceLocation(LevelUpHp.MOD_ID, name)
}
